pennylane/workflow/qnode.py

```diff
@@ -110,15 +110,33 @@ def _to_qfunc_output_type(
     if has_partitioned_shots:
         return tuple(_to_qfunc_output_type(r, qfunc_output, False) for r in results)
 
+    qfunc_output_leaves, qfunc_output_structure = qml.pytrees.flatten(
+        qfunc_output, is_leaf=lambda obj: isinstance(obj, (qml.measurements.MeasurementProcess))
+    )
+    num_of_measurements = len(qfunc_output_leaves)
+
+    results_leaves, results_structure = qml.pytrees.flatten(results)
+    num_of_results = len(results_leaves)
+
     # Special case of single Measurement in a list
     if isinstance(qfunc_output, list) and len(qfunc_output) == 1:
         results = [results]

```

Comment:
Do we need this logic anymore? The `qfunc_output_structure` should have the information about repacking the result leaves back into a list. 🤔